intel_extension_for_transformers/neural_chat/pipeline/plugins/retrieval/retrieval_agent.py

```diff
@@ -16,44 +16,44 @@
 # limitations under the License.
 
 import os
-from typing import List
-from .retrieval_base import Retriever
+from typing import Dict, List, Any
 from .detector.intent_detection import IntentDetector
-from .indexing.indexing import DocumentIndexing
+from .parser.parser import DocumentParser
 from intel_extension_for_transformers.neural_chat.pipeline.plugins.prompt.prompt_template \
     import generate_qa_prompt, generate_prompt, generate_qa_enterprise
-import logging
-logging.basicConfig(
-    format="%(asctime)s %(name)s:%(levelname)s:%(message)s",
-    datefmt="%d-%M-%Y %H:%M:%S",
-    level=logging.INFO
-)
+from intel_extension_for_transformers.langchain.embeddings import HuggingFaceEmbeddings, \
```

Comment:
do we need to import all these embeddings?

Reply:
yes, it would be more convenient to adjust different user-specific embedding model. We can also import the specific embedding according to the user-specific embedding model. But there is no time difference